scripts/build_osm_network.py

```diff
@@ -0,0 +1,1051 @@
+    line1 (dict): A dictionary representing the first line.
+    line2 (dict): A dictionary representing the second line.
+
+    Returns:
+    tuple: A tuple of boolean values indicating the connection status between
+    the lines.
+
+    The tuple contains four elements:
+    - True if the first line's bus_0_coors is almost equal to the second line's
+      bus_0_coors, False otherwise.
+    - True if the first line's bus_0_coors is almost equal to the second line's
+      bus_1_coors, False otherwise.
+    - True if the first line's bus_1_coors is almost equal to the second line's
+      bus_0_coors, False otherwise.
+    - True if the first line's bus_1_coors is almost equal to the second line's
+      bus_1_coors, False otherwise.
+    """
+    return (
+        are_almost_equal(line1["bus_0_coors"], line2["bus_0_coors"]),
+        are_almost_equal(line1["bus_0_coors"], line2["bus_1_coors"]),
+        are_almost_equal(line1["bus_1_coors"], line2["bus_0_coors"]),
+        are_almost_equal(line1["bus_1_coors"], line2["bus_1_coors"]),
+    )
+
+
+def _dfs(adj_matrix, visited, current_vertex, path):
+    """
+    Perform a depth-first search (DFS) on a graph represented by an adjacency
+    matrix.
+
+    Parameters:
+    - adj_matrix (list of lists): The adjacency matrix representing the graph.
+    - visited (list of bool): A list to keep track of visited vertices.
+    - current_vertex (int): The current vertex being visited.
+    - path (list): The path of vertices visited so far.
+
+    Returns:
+    - path (list): The path of vertices visited during the DFS.
+    """
+    visited[current_vertex] = True
+    path.append(current_vertex)
+    for neighbor in range(len(adj_matrix)):
+        if adj_matrix[current_vertex][neighbor] == 1 and not visited[neighbor]:
+            _dfs(adj_matrix, visited, neighbor, path)
+    return path
+
+
+# Returns all connected paths as a vector
+def find_paths(adj_matrix):
+    """
+    Find all paths in a graph represented by an adjacency matrix.
+
+    Parameters:
+    - adj_matrix (list of lists): The adjacency matrix representing the graph.
+
+    Returns:
+    - paths (list of lists): A list of lists, where each inner list represents
+      a path in the graph.
+    """
+    visited = [False] * len(adj_matrix)
+    paths = []
+    for vertex in range(len(adj_matrix)):
+        if not visited[vertex]:
+            path = _dfs(adj_matrix, visited, vertex, [])
+            if path:
+                paths.append(path)
+    return paths
+
+
+def are_almost_equal(point1, point2, tolerance=1e-6):
+    """
+    Check if two Shapely points are almost equal with a given tolerance.
+
+    Args:
+    point1 (Point): First Shapely point.
+    point2 (Point): Second Shapely point.
+    tolerance (float): Tolerance for coordinate deviation.
+
+    Returns:
+    bool: True if the points are almost equal, False otherwise.
+    """
+    return abs(point1.x - point2.x) < tolerance and abs(point1.y - point2.y) < tolerance
```

Comment:
this function can be substituted by `shapely` function: `point1.almost_equals(point2)`

Reply:
good point, code is outdated anyway, function not used.
done.